lib/clientv3/profile_e2e_test.go

```diff
@@ -250,6 +250,44 @@ var _ = testutils.E2eDatastoreDescribe("Profile tests", testutils.DatastoreEtcdV
 			_, outError = c.Profiles().Get(ctx, name2, options.GetOptions{})
 			Expect(outError).To(HaveOccurred())
 			Expect(outError.Error()).To(ContainSubstring("resource does not exist: Profile(" + name2 + ") with error:"))
+			By("Creating Profile (projectcalico-allow-all) and expecting an error")
```

Comment:
Assuming that [this assertion still passes](https://github.com/projectcalico/libcalico-go/pull/1207/files#diff-846336915a67afeb5c93349c4c88e4efR244-R247), then I think there is a bug in your code in the List() implementation (called out above as well).

We should now always expect a single profile on List() calls.

Reply:
Yep what I had wasn't quite right...